services/security/src/integration-tests/public/password-reset-token.spec.ts

```diff
@@ -0,0 +1,206 @@
+    return request(app)
+      .post("/api/users/password-reset-token")
+      .send({ username: "wrongUsername" })
+      .expect("Content-Type", /json/)
+      .expect(UNAUTHORIZED)
+      .expect(deepEqualOmit(BadRequestResponses.tokenMissingOrInvalid));
+  });
+
+  it("Should return bad request if a user wants reset password and won't provide a username", () => {
+    const { app } = GLOBAL.bootstrap;
+    return request(app)
+      .post("/api/users/password-reset-token")
+      .expect("Content-Type", /json/)
+      .expect(BAD_REQUEST)
+      .expect(deepEqualOmit({ error: '"username" is required' }));
+  });
+
+  it("Should return bad request if a user wants reset password and provide bad username", async () => {
+    const { authClient, superAdminUser, app } = GLOBAL.bootstrap;
+    const { accessToken } = await authClient.login(superAdminUser.username, superAdminUser.password);
+
+    return request(app)
+      .post("/api/users/password-reset-token")
+      .set("Authorization", `Bearer ${accessToken}`)
+      .send({ username: "bad-user-name" })
+      .expect("Content-Type", /json/)
+      .expect(NOT_FOUND)
+      .expect(deepEqualOmit(BadRequestResponses.wrongUsername));
+  });
+
+  it("Should reset password if user provide valid token", async () => {
+    const { authClient, superAdminUser, app } = GLOBAL.bootstrap;
+    const { accessToken } = await authClient.login(superAdminUser.username, superAdminUser.password);
+
+    const response = await request(app)
+      .post("/api/users/password-reset-token")
+      .set("Authorization", `Bearer ${accessToken}`)
+      .send({ username: normalUser.username })
+      .expect("Content-Type", /json/)
+      .expect(CREATED);
+
+    const { resetPasswordToken } = response.body;
+    assert(isNotEmptyString(resetPasswordToken));
+
+    return request(app)
+      .post(`/api/users/reset-password/${resetPasswordToken}`)
+      .send({ newPassword: "newValidPassword123" })
+      .expect("Content-Type", /json/)
+      .expect(CREATED);
+  });
+
+  it("Should set resetPasswordToken to null after successful password reset action", async () => {
+    const { authClient, superAdminUser, usersRepository, app } = GLOBAL.bootstrap;
+    const { accessToken } = await authClient.login(superAdminUser.username, superAdminUser.password);
+
+    const response = await request(app)
+      .post("/api/users/password-reset-token")
+      .set("Authorization", `Bearer ${accessToken}`)
+      .send({ username: normalUser.username })
+      .expect("Content-Type", /json/)
+      .expect(CREATED);
+
+    const { resetPasswordToken } = response.body;
+
+    await request(app)
+      .post(`/api/users/reset-password/${resetPasswordToken}`)
+      .send({ newPassword: "newValidPassword123" })
+      .expect("Content-Type", /json/)
+      .expect(CREATED);
+
+    const user = await usersRepository.findByUsername(normalUser.username);
+
+    assert.strictEqual(user?.resetPasswordToken, null);
```

Comment:
This can pass if `user` is null, while we're testing `resetPasswordToken` nullability, not `user`.

Reply:
There is optional operator (checking if the parent of a nested key is nullish - null or undefined).
`assert.strictEqual(undefined, null)` will return false.